### ISSUE TITLE
Can't create Dev Drives

### ISSUE THREAD

**Issue by DavidZidar** [microsoft/devhome#1622]
### Dev Home version

0.5

### Windows build number

10.0.22621.2361

### Steps to reproduce the bug

* Install Dev Home 0.1 through 0.5
* Install the latest Windows update KB5030310 which should enable Dev Drive
* Open Settings > System > Storage > Disks & volumes

### Expected result

I expect the feature to add a Dev Drive to be present.

### Actual result

The feature is missing.

**Comment by DavidZidar** [comment]
On 3 other computers I installed the Windows update first and then installed Dev Home and they all have the Dev Drive feature. The only computer that is missing the Dev Drive feature is the one computer that I want to use it on which is my main workstation where I installed Dev Home before the update.

I was never unable to actually run Dev Home until today though because of issue #797 which I figured out moments ago.

**Comment by bbonaby** [comment]
Hi @DavidZidar quick question. On your main workstation, in a terminal or cmd prompt, if you enter the command `fsutil devdrv query` what does the output say?

**Comment by DavidZidar** [comment]
@bbonaby It says "devdrv is an invalid parameter.", on my other computers where Dev Drives works it says "Developer volumes are enabled."

**Comment by bbonaby** [comment]
Thanks @DavidZidar, I see, looks like on your main machine the feature isn't fully lit up after the update. Would you be able to reboot the workstation one more time when you can, and then try again? Let me know if you still do not see the Dev Drive section in settings or if you still get the "devdrv is an invalid parameter." error again after that. Thanks!

**Comment by DavidZidar** [comment]
Sure, I rebooted once again (for like the 5th time today, haha). Still the same error.

**Comment by DavidZidar** [comment]
I've also tried to reset and repair the Dev Home app twice to no avail. I can't uninstall it to attempt to reinstall it because it seems to be a system app now.

**Comment by bbonaby** [comment]
Thanks for the update @DavidZidar, I created a bug in Feedback Hub to engage with our internal Storage team about this issue. As for being unable to uninstall the app for now, that would be because Dev Home is now a system component.

**Comment by DavidZidar** [comment]
Alright! I was looking forward to trying Dev Drive so thank you for looking into this. If there are any steps I could attempt to get it working I would be happy to try.

**Comment by joegasper** [comment]
@DavidZidar I was having the same issue. Seems like I was doing every correct, including dealing with temporary enterprise control features (Entra ID joined) and steps to enable Dev Drive (GPO/Intune). Too many reboots. 

Finally discovered I was missing [KB5030509](https://support.microsoft.com/en-us/help/5030509) - September 26, 2023—Windows configuration update (follow the instructions at the bottom).

I think [Controlled Feature Rollout (CFR](https://support.microsoft.com/en-us/topic/windows-configuration-updates-8c670410-7460-4c2b-a2b8-e6d58502507c)) got me.

I now have Dev Drive (and all the other goodies, like Windows Copilot). _fsutil devdrv query_ works; System, Storage, Drives & Volumes shows I can create a Dev Drive. All good now.

**Comment by DavidZidar** [comment]
Hmm, interesting. I'm not on Windows 11 Enterprise though, I use Windows 11 Pro and from what I can tell my Surface Pro laptop has the exact same list of updates applied as my workstation. Neither have KB5030509 in Update History and yet Dev Drive works fine on my laptop without it.

I have toggled "Get the latest updates as soon as they're available" on and checked for updates multiple times, the only optional updates I haven't installed are a bluetooth driver and a cumulative update for .NET Framework.

**Comment by DavidZidar** [comment]
I just tried enabling the Dev Drive group policy anyway but that didn't make a difference.

**Comment by DavidZidar** [comment]
Oh, you know what, the Update history UI was misleading me. My Surface Pro does indeed have the KB5030509 update listed under "Other Updates", but my workstation does not. I am at a loss on how to install it though. I have followed all the right steps as far as I know.

**Comment by joegasper** [comment]
I couldn't find KB5030509 in the Windows Update Catalog for a direct download - it just showed up for me first check for updates after enabling the "Get latest updates..." setting.  So maybe CFR - "Using CFR, we gradually roll out new features. When we have confirmed that each feature is ready, we will gradually roll it out to other devices. Eventually, it will be included in a later monthly update."

**Comment by DavidZidar** [comment]
Thanks for looking. I guess I'll just have to wait. I really, really dislike gradual rollouts. Clicking check for updates repeatedly should bypass that I think, but that's just me.

**Comment by DavidZidar** [comment]
Just installed the latest cumulative updates and I still can't create dev drives and I still haven't gotten the configuration update. 😢

**Comment by TroutZhang** [comment]
I met the same issue, and thanks to @joegasper 
Got it figured out after installing the Sept 26 Windows Configure Update https://support.microsoft.com/en-us/topic/september-26-2023-windows-configuration-update-542780c2-594c-46cb-979d-11116fe164ba

**Comment by DavidZidar** [comment]
Yes, but how do you install this update? I've followed the instructions on that page but I'm not getting the update no matter how many times I try.

**Comment by hamnarauf** [comment]
@DavidZidar In my case, I enabled the "Get latest updates as soon as they are available" and clicked "Check for updates". This showed me a pending update which required a restart. After Update and Restart my PC, I got the option to create Dev Drive.

![image](https://github.com/microsoft/devhome/assets/77397009/f85fa9bb-f2cb-4d50-8a26-6e2f6518046f)

Hope this helps!

**Comment by DavidZidar** [comment]
That is exactly what I have done too, still not getting the update.

**Comment by Dolliwyx** [comment]
I'm also currently having this issue on Windows 11 Pro Insider Preview 22H2, build 23541.1000. I've tried the following as well:
- Enabled Dev Drive through the Local Policy Editor.
- Ticked "Get the latest updates as soon as they are available" and checked for updates."
- Rebooted at least a few times

However, I still don't have the option to create Dev Drives. Running `fsutil devdrv query` also gives out "devdrv is an invalid parameter.". It also seems that I do not have the KB5030509 update installed in my system, regardless of the steps I've taken above.

**Comment by AdamBraden** [comment]
> I'm also currently having this issue on Windows 11 Pro Insider Preview 22H2, build 23541.1000. I've tried the following as well:
> 
> * Enabled Dev Drive through the Local Policy Editor.
> * Ticked "Get the latest updates as soon as they are available" and checked for updates."
> * Rebooted at least a few times
> 
> However, I still don't have the option to create Dev Drives. Running `fsutil devdrv query` also gives out "devdrv is an invalid parameter.". It also seems that I do not have the KB5030509 update installed in my system, regardless of the steps I've taken above.

@Dolliwyx - your build is an older Dev External channel. The KBs mentioned above do not apply. Can you confirm if your device is enterprise/IT managed with Intune or configmgr? If so, then you need your IT to enable Dev Drive volumes for your organization. See https://learn.microsoft.com/en-us/windows/dev-drive/group-policy

**Comment by AdamBraden** [comment]
@DavidZidar - I'm checking with the servicing team to see how best to figure out why your device is not getting the Configuration Update Package.

**Comment by Dolliwyx** [comment]
> @Dolliwyx - your build is an older Dev External channel. The KBs mentioned above do not apply. Can you confirm if your device is enterprise/IT managed with Intune or configmgr? If so, then you need your IT to enable Dev Drive volumes for your organization. See https://learn.microsoft.com/en-us/windows/dev-drive/group-policy

Hey! Upon checking, I don't think my device is managed by an organization as I don't have Intune or configmgr in my computer, just the Local Group Policy Editor. Another thing to note, my school account is also linked to my device at the moment, but I don't think my university can manage my own device. But I have removed it for the meantime, could that have been a factor? Thank you for your help @AdamBraden !

**Comment by DavidZidar** [comment]
@AdamBraden I just installed the latest Cumulative Update Preview (KB5031455) that was released yesterday and all of a sudden the configuration update (KB5030509) was installed too so now the Dev Drive feature is enabled! Thank you for checking with the servicing team, looks like I won't be needing more assistance but I appreciate the help all the same!

@Dolliwyx Maybe this update will resolve your issue as well?
